assets/js/view/part/base.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'snabberb/component'
+
+module View
+  module Part
+    class Base < Snabberb::Component
+      needs :region_use
+      needs :tile, default: nil
+
+      CENTER = [7, 8, 9, 14, 15, 16].freeze
+
+      LEFT_CENTER = [6, 7, 13, 14].freeze
+      RIGHT_CENTER = [9, 10, 16, 17].freeze
+
+      LEFT_CORNER = [5, 12].freeze
+      RIGHT_CORNER = [11, 18].freeze
+
+      def preferred_render_locations
+        [
+          {
+            region_weights: {},
+            x: 0,
+            y: 0,
+          }
+        ]
+      end
+
+      def translate
+        x = render_location[:x]
+        y = render_location[:y]
+
+        "translate(#{x} #{y})"
+      end
+
+      def increment_cost
+        region_weights = render_location[:region_weights]
+        region_weights = region_weights.keys.flatten unless region_weights.is_a?(Array)
+
+        region_weights.each do |region|
+          @region_use[region] += 1
```

Comment:
do you want to increment by weight here or static 1?

Reply:
At first I had it incrementing by weight here, then I moved the weight to where the render location is being decided. I thought about incrementing in both but figured it would then make sense to have two weights--one for how much a part cares about what was rendered in that region before, and how much the part uses that region, but wasn't sure the extra complexity was worthwhile.

Reply:
0f2c9d1